Source/model/draw/DropShadow.swift

```diff
@@ -1,16 +1,22 @@
 import Foundation
 
+public enum EffectSource: String {
+    case SourceGraphic, SourceAlpha, BackgroundImage, BackgroundAlpha, FillPaint, StrokePaint
+}
+
 open class DropShadow: Effect {
 
-    open let radius: Double
-    open let offset: Point
-    open let color: Color
-    open let input: Effect?
+    open var radius: Double
+    open var offset: Point?
+    open var color: Color
+    open var input: Effect?
+    open var source: EffectSource
```

Comment:
Need to replace `var` with `let`

Reply:
Process of filter parsing is already really complicated thanks to all the 'in' and 'result' chains. Replacing var with let will complicate it even further. Are you sure it's necessary?

Reply:
I'm sure) Making model mutable will break the whole Macaw concept.